provider/views.py

```diff
@@ -261,12 +263,22 @@ def handle(self, request, post_data=None):
 
         authorization_form = self.get_authorization_form(request, client,
             post_data, data)
-
-        if not authorization_form.is_bound or not authorization_form.is_valid():
-            return self.render_to_response({
-                'client': client,
-                'form': authorization_form,
-                'oauth_data': data, })
+        
+        if data.get('response_type', None):
+            if not authorization_form.is_bound \
+                or not authorization_form.is_valid():
```

Comment:
no need to break this line.

optional: not (x and y) would read a little better.